lib/Recovery.php

```diff
@@ -221,6 +221,11 @@ private function addRecoveryKeys($path) {
 		$dirContent = $this->view->getDirectoryContent($path);
 		foreach ($dirContent as $item) {
 			$filePath = $item->getPath();
+			// only add key for files owned by the user
+			if($item->getStorage()->instanceOfStorage('OC\Files\Storage\Shared')) {
```

Comment:
`ISharedStorage` to also exclude incoming federated shares.

See and also please review: https://github.com/owncloud/core/pull/29724

Reply:
And we can use ::class`

Reply:
hedcoding because we dont want inter app dependencies. might give problems during class loading in case files_sharing is disabled